cleaner.py

```diff
@@ -31,79 +28,73 @@ def chunks(l, n):
 
+        print('Delete all your messages in')
         for i, group in enumerate(groups):
-            print(f'{i+1}. {group.chat.title}')
+            print(f'  {i+1}. {group.title}')
+        print(f'  {len(groups) + 1}. (!) ALL OF THE ABOVE (!)\n')
 
-        print('')
-
-        group_n = int(input('Insert group number: '))
-        if group_n not in range(1, len(groups)+1):
-            print('Invalid group number. Exiting...')
+        n = int(input('Insert option number: '))
+        if not 1 <= n <= len(groups) + 1:
+            print('Invalid option selected. Exiting...')
             exit(-1)
 
-        selected_group = groups[group_n - 1]
-
-        selected_group_peer = app.resolve_peer(selected_group.chat.id)
-        self.peer = selected_group_peer
-        self.chat_id = selected_group.chat.id
-
-        print(f'Selected {selected_group.chat.title}\n')
-
-        return selected_group, selected_group_peer
+        self.chats = groups if n == len(groups) + 1 else [groups[n - 1]]
+ 
+        print(f'\nSelected {", ".join(c.title for c in self.chats)}.\n')
 
     def run(self):
-        while True:
-            q = self.search_messages()
-            self.update_ids(q)
-            messages_count = len(q['messages'])
-            print(f'Found {messages_count} of your messages in selected group')
-            if messages_count < self.search_limit:
-                break
-            self.add_offset += self.search_limit
-
-        self.delete_messages()
-
-    def update_ids(self, query: ChannelMessages):
-        for msg in query.messages:
-            self.message_ids.append(msg.id)
-
-        return len(query.messages)
-
-    def delete_messages(self):
-        print(f'Deleting {len(self.message_ids)} messages with message IDs:')
-        print(self.message_ids)
-        for message_ids_chunk in self.chunks(self.message_ids, 100):
+        for chat in self.chats:
+            peer = app.resolve_peer(chat.id)
+            message_ids = []
+            add_offset = 0
+
+            while True:
+                q = self.search_messages(peer, add_offset)
+                message_ids.extend(msg.id for msg in q['messages'])
+                messages_count = len(q['messages'])
+                print(f'Found {messages_count} of your messages in "{chat.title}"')
+                if messages_count < self.search_limit:
+                    break
+                add_offset += self.search_limit
+
+            self.delete_messages(chat.id, message_ids)
+
+    def delete_messages(self, chat_id, message_ids):
+        print(f'Deleting {len(message_ids)} messages with message IDs:')
+        print(message_ids)
+        for message_ids_chunk in self.chunks(message_ids, 100):
             try:
-                app.delete_messages(chat_id=self.chat_id,
-                                    message_ids=message_ids_chunk)
+                app.delete_messages(
+                    chat_id=chat_id,
+                    message_ids=message_ids_chunk)
```

Comment:
Last parenthesis on the next line please or move first argument to previous line;

```python
app.delete_messages(chat_id=chat_id,
                    message_ids=message_ids_chunk)
```

```python
app.delete_messages(
    chat_id=chat_id,
    message_ids=message_ids_chunk
)
```

Reply:
I think you're referring to an ealier commit. It's on a single line now.